### DISCUSSION TITLE
How to implement Keycloak with Saleor?

### DISCUSSION THREAD

**Discussion by maiaai** [saleor/saleor#8771] [General]
Hi, I want to replace existing saleor jwt system with keycloak: https://www.keycloak.org/. Is this possible? I got it working with pure Django project, but when I try to implement it within Saleor, I get issues that `AttributeError: 'WSGIRequest' object has no attribute 'user' ` .Also tried using django-keycloak package, and django-allauth since it also provides support for keycloak, but no luck. The issue is the same.
@karolkielecki

**Comment by karolkielecki** [comment]
@maiaai Hi. I think you can develop a [plugin](https://docs.saleor.io/docs/3.0/developer/extending/plugins) in that case. We developed the OpenID Connect (OIDC) plugin, but it's only available on Cloud as stated in our documentation.

**Comment by patrys** [comment]
Hi, we're open-sourcing the OIDC plugin which should allow you to integrate with Keycloak: #9406
